crates/assembly/src/linker/mod.rs

```diff
@@ -986,30 +768,48 @@ impl Linker {
     }
 
     /// Resolve a [Path] to a [ModuleIndex] in this graph
-    pub fn find_module_index(&self, name: &Path) -> Option<ModuleIndex> {
-        self.modules
-            .iter()
-            .position(|m| m.as_ref().is_some_and(|m| name == m.path()))
-            .map(ModuleIndex::new)
+    pub fn find_module_index(&self, path: &Path) -> Option<ModuleIndex> {
+        self.modules.iter().position(|m| path == m.path()).map(ModuleIndex::new)
     }
 
     /// Resolve a [Path] to a [Module] in this graph
-    pub fn find_module(&self, name: &Path) -> Option<ModuleLink> {
-        self.modules
-            .iter()
-            .find(|m| m.as_ref().is_some_and(|m| name == m.path()))
-            .cloned()
-            .unwrap_or(None)
+    pub fn find_module(&self, path: &Path) -> Option<&LinkModule> {
+        self.modules.iter().find(|m| path == m.path())
+    }
+}
+
+/// Const evaluation
+impl Linker {
+    /// Evaluate `expr` to a concrete constant value, in the context of the given item.
+    pub(super) fn const_eval(
+        &self,
+        gid: GlobalItemIndex,
+        expr: &ast::ConstantExpr,
+        cache: &mut ResolverCache,
+    ) -> Result<ast::ConstantValue, LinkerError> {
+        let symbol_resolver = SymbolResolver::new(self);
+        let mut resolver = Resolver {
+            resolver: &symbol_resolver,
+            cache,
+            current_module: gid.module,
+        };
+
+        ast::constants::eval::expr(expr, &mut resolver).map(|expr| expr.expect_value())
```

Comment:
If the const expression cannot be fully evaluated, this will panic rather than return an error. Is this the behavior we want, when we have the opportunity to signal the user how they messed up and recover?